inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/messages/request/arc/CreateArcRequest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Technische Universität Darmstadt under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The Technische Universität Darmstadt
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.experimental.api.messages.request.arc;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.VID;
+
+public class CreateArcRequest
+{
+    private String clientName;
+    private long projectId;
+    private long documentId;
+    private VID sourceId;
+    private VID targetId;
+    private String layer;
+
+    public String getClientName()
+    {
+        return clientName;
+    }
+
+    public void setClientName(String aClientName)
+    {
+        clientName = aClientName;
+    }
+
+
+    public long getProjectId()
+    {
+        return projectId;
+    }
+
+    public void setProjectId(long aProjectId)
+    {
+        projectId = aProjectId;
+    }
+
+    public long getDocumentId()
+    {
+        return documentId;
+    }
+
+    public void setDocumentId(long aDocumentId)
+    {
+        documentId = aDocumentId;
+    }
+
+    public VID getSourceId()
+    {
+        return sourceId;
+    }
+
+    public void setSourceId(VID aSourceId)
+    {
+        sourceId = aSourceId;
+    }
+
+    public VID getTargetId()
+    {
+        return targetId;
+    }
+
+    public void setTargetId(VID aTargetId)
+    {
+        targetId = aTargetId;
+    }
+    
+    public String getLayer()
+    {
+        return layer;
+    }
+
+    public void setLayer(String aLayer)
```

Comment:
Yes thats right, the only issue is, that the client only knows about the layers uiNames because it has no clue about any IDs. And the uiName of a layer can easily be retrieved e.g. from the dropdown. Otherwise, the client needs to remember all IDs which would introduce another state.

Reply:
The client needs to work with the IDs. The must never be used for logic, only for presentation. The IDs are way more stable than the UI names. So you need to be able to have both in the client, ID and UI name.